core-web/libs/sdk/client/README.md

```diff
@@ -169,237 +194,360 @@ const { pageAsset, content } = await client.page.get('/about-us', {
-    .fetch();
+import { createDotCMSClient } from '@dotcms/client/next';
+
+const client = createDotCMSClient({
+    dotcmsUrl: 'https://your-dotcms-instance.com',
+    authToken: 'your-auth-token',
+    siteId: 'your-site-id',
+    requestOptions: { cache: 'default' }
+});
 ```
 
-#### Setting Depth for Relationships
+2. Limit GraphQL queries to needed fields
+3. Use pagination for large collections
+4. Consider using the Content API separately for data not needed immediately
+
+### How to correctly type a page response?
 
-To fetch content with a specific relationship depth, use the `depth()` method:
+When using TypeScript, you can leverage generic types to get proper typing for your page and content data:
 
 ```typescript
-const depthResponse = await client.content
-    .getCollection('Blog')
-    .depth(2) // Fetch related content up to depth 2
-    .fetch();
+// Import the base DotCMS types
+import { DotCMSPageAsset } from '@dotcms/types';
+
+// Define the page structure by extending the base DotCMSPageAsset
+interface AboutUsPage extends DotCMSPageAsset {
+    vanityUrl: {
+        url: string;
+    }
+}
+
+// Define interfaces for your content types
+interface BlogPost {
+    title: string;
+    identifier: string;
+    urlTitle: string;
+    blogContent: {
+        json: string;
+    };
+}
+
+interface TeamMember {
+    name: string;
+    position: string;
+    bio?: string;
+}
+
+// Define the content response structure
+interface AboutUsContent {
+    blogPosts: BlogPost[];
+    teamMembers: TeamMember[];
+}
+
+// Use the type parameters to get fully typed responses
+const response = await client.page.get<{ pageAsset: AboutUsPage; content: AboutUsContent; }>('/about-us', {
+    languageId: '1',
+    mode: 'PREVIEW_MODE',
+    graphql: {
+        page: `
+            title
+            pageId
+            vanityUrl {
+                url
+            }
+        `,
+        content: {
+            blogPosts: `
+                BlogCollection(limit: 3) {
+                    title
+                    identifier
+                    ...blogFragment
+                }
+            `,
+            teamMembers: `
+                TeamMemberCollection(limit: 5) {
+                    name
+                    position
+                    bio
+                }
+            `
+        },
+        fragments: [
+            `
+                fragment blogFragment on Blog {
+                    urlTitle
+                    blogContent {
+                        json
+                    }
+                }
+            `
+        ]
+    }
+});
+
+const { pageAsset, content } = response;
+
+// Now you get full TypeScript support
+console.log(pageAsset.vanityUrl.url);        // TypeScript knows this exists
+console.log(content.blogPosts[0].title);     // TypeScript knows this exists
+console.log(content.teamMembers[0].position); // TypeScript knows this exists
 ```
 
-#### Combining Multiple Methods
+### How do I build complex content queries?
```

Comment:
Not sure, we need to be clever on provide the edge cases where you will use this instead of page, because people will be confused, thinking they have to use this.

Reply:
I'll remove this section